scripts/refresh-submodules.sh

```diff
@@ -25,12 +25,27 @@ submodules=(
 for ent in "${@:-${submodules[@]}}"; do
     submodule=${ent%%:*}
     [ ${submodule} == ${ent} ] && branch="master" || branch=${ent#*:}
-    GIT_DIR="$submodule/.git" git pull --ff-only origin ${branch}
-    SUMMARY=$(git submodule summary $submodule)
-    if grep '^ *<' <<< "$SUMMARY"; then
+
+    pushd .
+    cd $submodule
+    git fetch origin ${branch}
+    if ! git merge-base --is-ancestor HEAD origin/master; then
         echo "Non fast-forward changes detected! Fire three red flares from your flare pistol."
         exit 1
     fi
+    # collect the summary
+    head=$(git rev-parse --short=8 HEAD)
+    master=$(git rev-parse --short=8 master)
```

Comment:
You should use ${branch} instead of "master", as in the "git fetch" above and as in the old code. See commit cb4fe65a780bc6f14e6bd09210450aa1863f9019 by @xemul.

Reply:
ahh, right. i missed it. corrected.